chia/wallet/coin_selection.py

```diff
@@ -74,8 +74,24 @@ async def select_coins(
             smaller_coin_sum += coin.amount
             smaller_coins.append(coin)
     if smaller_coin_sum == amount:
-        log.debug(f"Selected all smaller coins because they equate to an exact match of the target.: {smaller_coins}")
-        return set(smaller_coins)
+        if len(smaller_coins) < 500:
+            log.debug(
+                f"Selected all smaller coins because they equate to an exact match of the target.: {smaller_coins}"
+            )
+            return set(smaller_coins)
+        else:
+            target_coins: Set[Coin] = set()

```

Comment:
this code does not do anything because we already know that `smaller_coin_sum == amount.`. We will never reach the break.

Also, it's complex.

Instead we should fall back to selecting the greater coin.

Reply:
Ok, i will select the greatest individual coin.